addons/mail/static/src/core/common/message.js

```diff
@@ -496,6 +496,132 @@ export class Message extends Component {
         this.state.showTranslation =
             !this.state.showTranslation && Boolean(message.translationValue);
     }
+
+    async fetchAuthorFromSuggestedRecipients() {
+        if (this.message.author) {
+            return this.message.author;
+        }
+        // auto-create partners of sender from suggested partners
+        const authorCheck = this.message.thread.suggestedRecipients.find((recipient) =>
+            recipient.email.includes(this.message.email_from)
+        );
+        const recipientEmails = [authorCheck.email];
+        const partners = await rpc("/mail/partner/from_email", {
+            emails: recipientEmails,
+            additional_values: { [authorCheck.email]: authorCheck.create_values || {} },
+        });
+        let author;
+        for (const index in partners) {
+            const partnerData = partners[index];
+            const persona = this.store.Persona.insert({ ...partnerData, type: "partner" });
+            const email = recipientEmails[index];
+            const recipient = this.message.thread.suggestedRecipients.find(
+                (recipient) => recipient.email === email
+            );
+            Object.assign(recipient, { persona });
+            author = persona;
+        }
```

Comment:
Why do you need to do all of this? I have so much questions. If you have the email from your search above, couldn't you stop then? `recipientEmails` is a list containing a singular email and you never push anything inside it, why?

Reply:
Hello
This function is especially for incoming emails, where message_type is `email`, there is no partner when there is an incoming email even in `suggestedRecipients` we only have name and email.
This function creates the partner or finds an existing partner (if any) and is used as `To:` in the reply-all dialog.
![image](https://github.com/user-attachments/assets/2d7ff544-e4b8-47ba-8e55-fbda3c6d918f)

Reply:
>I get that this is what you are doing here, but why like this?
>
> recipientEmails is a list with only one email that's never updated, why a list then?

Actually `/mail/partner/from_email` accepts `emails` to be in the form of a list.
So I kept that part as a list. But I guess it is not readable
Alright making it simple now.

---
  > The day you have multiple partners for the same email you will encounter a crash in the loop. How do you handle this case?

Isn't that `_find_or_create_from_emails` returns a single partner for one email